### PR TITLE
Pull Fields instance once from LeafReader in completion stats

### LINKED ISSUE #6593: Completion stats walk through all fields
To compute completion stats we currently walk through all the lucene fields and all segments in our shards. This is inefficient as we only need to walk through fields used for completion. This becomes even more extreme for people not using the completion suggester.

Relevant code: 
https://github.com/elasticsearch/elasticsearch/blob/master/src/main/java/org/elasticsearch/search/suggest/completion/Completion090PostingsFormat.java#L360

### PR DESCRIPTION
**Pull request by mikemccand**
Closes #6593

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/search/suggest/completion/CompletionFieldStats.java [hunk 1]
```diff
@@ -32,28 +34,35 @@
...
                         long fstSize = ((CompletionTerms) terms).suggester().ramBytesUsed();
-                        if (fields != null && fields.length > 0 && Regex.simpleMatch(fields, fieldName)) {
+                        if (fieldNamePatterns != null && fieldNamePatterns.length > 0 && Regex.simpleMatch(fieldNamePatterns, fieldName)) {
                             completionFields.addTo(fieldName, fstSize);
                         }
                         sizeInBytes += fstSize;
                     }
                 }
-            } catch (IOException ignored) {
-                throw new ElasticsearchException(ignored);
```
Comment (bleskes) [inline core/src/main/java/org/elasticsearch/search/suggest/completion/CompletionFieldStats.java]: :)

**Comment by bleskes** [comment]
LGTM

File: core/src/main/java/org/elasticsearch/search/suggest/completion/CompletionFieldStats.java (hunk 1 above)
Comment (mikemccand) [inline core/src/main/java/org/elasticsearch/search/suggest/completion/CompletionFieldStats.java]: Yes :)

**Comment by areek** [comment]
FYI @mikemccand, I backported this change to 2.x

**Pull request merged**